node-test/src/test/scala/org/bitcoins/node/NeutrinoNodeWithWalletTest.scala

```diff
@@ -184,10 +192,6 @@ class NeutrinoNodeWithWalletTest extends NodeUnitTest {
       val output = TransactionOutput(sats, spk)
 
       for {
-        _ <- node.sync()
-        _ <- NodeTestUtil.awaitSync(node, bitcoind)
-        _ <- NodeTestUtil.awaitCompactFiltersSync(node, bitcoind)
```

Comment:
I am assuming there will be CI failures because we don't wait for these to be done syncing

Reply:
Syncing is now done here: https://github.com/bitcoin-s/bitcoin-s/pull/2461/files#diff-7075da7fcab619a015363e10a0f880b3b76b03e6b3cfd8daaf07ccbe53448f3cR584

I am running into some time out issues locally though wrt to `_ <- AsyncUtil.awaitConditionF(condition1)`

Reply:
It starts syncing, but it will start running the test before it is done because it doesn't have the 
```
        _ <- NodeTestUtil.awaitSync(node, bitcoind)
        _ <- NodeTestUtil.awaitCompactFiltersSync(node, bitcoind)
```

Reply:
:man_facepalming:  [insert I am not a smart man forrest gump meme]

However this is uncovering another problem, our block count and filter count never seem to converge 

>[info] NeutrinoNodeWithWalletTest:
blockCount=101 filterCount=99
blockCount=101 filterCount=99
blockCount=101 filterCount=99
blockCount=101 filterCount=99
blockCount=101 filterCount=99
blockCount=101 filterCount=99
blockCount=101 filterCount=99
blockCount=101 filterCount=99

Reply:
FWIW, this is the initial bug that blocked me on #2436 and I thought i had solved it, but I guess not. I'll keep digging.

Reply:
So the problem with this seems to be executing callbacks. We are hanging here forever

https://github.com/bitcoin-s/bitcoin-s/blob/0d4af2f83df4d247f9885c09d5c58d11e77c7178/node/src/main/scala/org/bitcoins/node/networking/peer/DataMessageHandler.scala#L129

Reply:
So the problem is this: 

We have a race condition between `walletF` and the nodeCallbacks being setup. 

Since we moved sync logic into the fixture creation, we cannot register wallet callbacks anymore from initial sync (we do not have access to a wallet!). 

We were getting around this before with 

```scala
  private var walletP: Promise[Wallet] = Promise()
  private var walletF: Future[Wallet] = walletP.future
```

and then everytime we would start a test case, we would complete the `Promise` with the wallet in the given to us by the fixture. 

```scala
      val NeutrinoNodeFundedWalletBitcoind(node, wallet, bitcoind, _) = param

      walletP.success(wallet)
```

and finally, after that promise is completed, we can actually execute our callbacks as defined

```scala
    val onCompactFilters: OnCompactFiltersReceived = { blockFilters =>
      for {
        wallet <- walletF
        _ <- wallet.processCompactFilters(blockFilters)
      } yield ()
    }
```

However now since we are doing sync logic in the fixture, we can no longer have the callbacks registered when the fixture is being started, else we will hang forever.